tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/hlo_canonicalize_reduction.cc

```diff
@@ -0,0 +1,238 @@
+// This file canonicalize reduction ops in hlo dialect to match the
```

Comment:
Please add the standard TensorFlow license in source files.

Reply:
Done